### DISCUSSION TITLE
Is it possible to add custom colors to the theme?

### DISCUSSION THREAD

**Discussion by EugenEistrach** [saadeghi/daisyui#1293] [Q&A] [answered]
Hello,
currently trying out DaisyUI and hit a small roadblock. 
I have couple of social buttons on my page (github, discord, google, etc..). 
Now they need to use different colors than the ones provided in the theme by default. 
Now lets say I want to reuse the 'btn' class but I want to have custom colors like for example btn-discord or btn-google. 
Is it possible somehow?

Obviously I could just create the classes myself with @apply always but I'm wondering if there is some way to have those classes generated the same way as the default classes (btn-primary, btn-secondary, etc.)?

**Comment by saadeghi** [comment, marked as answer]
There are 3 ways:
1. Write some custom CSS using `@apply` as you mentioned.
2. Change the colors of a button using class names. This is the easiest way when you only need that new color a few times and only for the button:
  https://play.tailwindcss.com/cVVNyXq6bA
    ```html
    <button class="btn bg-[#1EA1F1] border-[#1EA1F1] hover:bg-[#1C96E1] hover:border-[#1C96E1]">Twitter</button>
    ```
4. Add the a new class name for the button on the theme config. This is useful when you will use the new button multiple times. You can also use different colors for different themes:
  https://play.tailwindcss.com/jl4NPeFyHk?file=config
    ```js
    module.exports = {
      plugins: [require('daisyui')],
      daisyui: {
        themes: [
          {
            light: {
              ...require('daisyui/src/colors/themes')['[data-theme=light]'],
              '.btn-twitter': {
                'background-color': '#1EA1F1',
                'border-color': '#1EA1F1',
              },
              '.btn-twitter:hover': {
                'background-color': '#1C96E1',
                'border-color': '#1C96E1',
              },
            },
          },
        ],
      },
    }
    ```

**Reply by Sleepful** [reply]
This should be in the docs! :D

**Comment by Sleepful** [comment]
you can simply append css custom properties to the themes that you are using, then you can use something like `@layer` to create a tailwind class that will use your new variable:

```css
@layer base {
  .bg-input {
    background-color: hsl(var(--bi)) !important;
  }
}

[data-theme=cupcake] {
  --bi: 100 100% 100%;
}

[data-theme=dark] {
  --bi: 220 17% 17%;
}
```
